skll/learner/__init__.py

```diff
@@ -1151,41 +1186,44 @@ def train(
         return grid_score, grid_cv_results
 
     def evaluate(
-        self, examples, prediction_prefix=None, append=False, grid_objective=None, output_metrics=[]
-    ):
+        self,
+        examples: FeatureSet,
+        prediction_prefix: Optional[str] = None,
+        append: bool = False,
+        grid_objective: Optional[str] = None,
+        output_metrics: List[str] = [],
+    ) -> EvaluateTaskResults:
         """
         Evaluate the learner on a given dev or test ``FeatureSet``.
 
         Parameters
         ----------
-        examples : skll.FeatureSet
+        examples : skll.data.FeatureSet
             The ``FeatureSet`` instance to evaluate the performance of the
             model on.
-        prediction_prefix : str, optional
+        prediction_prefix : Optional[str]
             If not ``None``, predictions will also be written out to a file with
             the name  ``<prediction_prefix>_predictions.tsv``. Note that
             the prefix can also contain a path.
             Defaults to ``None``.
-        append : bool, optional
-            Should we append the current predictions to the file if
-            it exists?
+        append : bool
+            Should we append the current predictions to the file if it exists?
             Defaults to ``False``.
-        grid_objective : function, optional
-            The objective function that was used when doing
-            the grid search.
+        grid_objective : Optional[str]
+            The objective function that was used when doing the grid search.
             Defaults to ``None``.
-        output_metrics : list of str, optional
-            List of additional metric names to compute in
-            addition to grid objective. Empty by default.
+        output_metrics : List[str]
+            List of additional metric names to compute in addition to grid
+            objective. Empty by default.
             Defaults to an empty list.
 
         Returns
         -------
-        res : 6-tuple
-            The confusion matrix, the overall accuracy, the per-label
-            PRFs, the model parameters, the grid search objective
+        EvaluateTaskResults
+            A 6-tuple containing the confusion matrix, the overall accuracy,
+            the per-label PRFs, the model parameters, the grid search objective
             function score, and the additional evaluation metrics, if any.
-            For regressors, the first two elements are ``None``.
+            For regressors, the first two elements in the tuple are ``None``.
         """
```

Comment:
Is mutability the motivation behind having this be a tuple rather than a dictionary? If not, a dict seems like a friendlier option.

Reply:
Yep, again, with the benefit of hind-sight, we should have used a dictionary but things are what they are and I don't want to break things too much at this point :)